src/main/java/by/itech/lab/supplier/controller/UserController.java

```diff
@@ -75,14 +75,14 @@ public UserDto getUseByUsername(@PathVariable String username) {
         log.debug("request to get User : {}", username);
         return userService.findByUsername(username);
     }
-    @AdminAccess
+
     @ResponseStatus(HttpStatus.OK)
     @PutMapping(ApiConstants.URL_ID_PARAMETER + ApiConstants.URL_PASSWORD_PARAMETER)
     public int changePassword(@PathVariable Long id, @RequestBody String password) {
         return userService.changePassword(id, password);
```

Comment:
You should be protecting this method, but not @AdminAccess. Now any user can change the password to any user of the company which owns. But the user must have opportunity to change only his password.